tests/unit/plugins/modules/test_ec2_dedicated_host.py

```diff
@@ -0,0 +1,100 @@
+
+from ansible_collections.amazon.aws.plugins.modules import ec2_dedicated_host
+
+module_name = "ansible_collections.amazon.aws.plugins.modules.ec2_dedicated_host"
+
+
+@pytest.fixture(name="ansible_module")
+def fixture_ansible_module():
+    module = MagicMock()
+    module.params = {}
+    module.exit_json = MagicMock()
+    module.fail_json = MagicMock()
+    module.fail_json.side_effect = SystemExit(1)
+    return module
+
+
+@pytest.mark.parametrize(
+    "dedicated_hosts,expected",
+    [
+        ([], None),
+        (
+            [{"host_id": "h-01ca0a554a7f8787b", "mac_os_latest_supported_versions": ["15.3", "14.7.3", "13.7.3"]}],
+            {"host_id": "h-01ca0a554a7f8787b", "mac_os_latest_supported_versions": ["15.3", "14.7.3", "13.7.3"]},
+        ),
+    ],
+)
+@patch(module_name + ".describe_ec2_dedicated_hosts")
+def test_get_ec2_dedicated_host_with_lookup_host_id(
+    m_describe_ec2_dedicated_hosts, ansible_module, dedicated_hosts, expected
+):
+    ansible_module.params = {"host_id": "h-012345", "lookup": "host_id"}
+    client = MagicMock()
+    m_describe_ec2_dedicated_hosts.return_value = dedicated_hosts
+
+    assert ec2_dedicated_host.get_ec2_dedicated_host(client, ansible_module) == expected
+    expected_params = {
+        "Filter": [{"Name": "state", "Values": ["available", "under-assessment", "permanent-failure"]}],
+        "HostIds": ["h-012345"],
+    }
+    m_describe_ec2_dedicated_hosts.assert_called_once_with(client, **expected_params)
+
+
+@pytest.mark.parametrize(
+    "dedicated_hosts,expected",
+    [
+        ([], None),
+        ([{"host_id": "host-1"}, {"host_id": "host-2", "Tags": {"foo": "bar"}}], None),
+        ([{"host_id": "host-1"}, {"host_id": "host-2", "Tags": {"phase": "test"}}], None),
+        (
+            [
+                {"host_id": "host-1"},
+                {"host_id": "host-2", "Tags": {"foo": "bar", "phase": "test", "session": "active"}},
+            ],
+            None,
+        ),
+        (
+            [{"host_id": "host-1"}, {"host_id": "host-2", "Tags": {"foo": "bar", "phase": "test"}}],
+            {"host_id": "host-2", "Tags": {"foo": "bar", "phase": "test"}},
+        ),
+        (
+            [
+                {"host_id": "host-1", "Tags": {"foo": "bar", "phase": "test"}},
+                {"host_id": "host-2", "Tags": {"foo": "bar", "phase": "test"}},
+            ],
+            [
+                {"host_id": "host-1", "Tags": {"foo": "bar", "phase": "test"}},
+                {"host_id": "host-2", "Tags": {"foo": "bar", "phase": "test"}},
+            ],
+        ),
+    ],
+)
+@patch(module_name + ".boto3_tag_list_to_ansible_dict")
+@patch(module_name + ".describe_ec2_dedicated_hosts")
+def test_get_ec2_dedicated_host_with_lookup_tag(
+    m_describe_ec2_dedicated_hosts, m_boto3_tag_list_to_ansible_dict, ansible_module, dedicated_hosts, expected
+):
+    ansible_module.params = {"host_id": "h-012345", "lookup": "tag", "tags": {"foo": "bar", "phase": "test"}}
```

Comment:
Can we add a test for invalid lookup value? Something like - `ansible_module.params = {"lookup": "invalid_value"}`

Reply:
This is not a valid test, the `lookup` argument has a limited number of accepted values `tag, host_id`

Reply:
I agree. I was just looking for a negative test. It was merely a suggestion, not a requirement to add it.